### PR TITLE
Move logs and permission such that Apache user can write to logs

### PR DESCRIPTION
**Pull request by sleitner**
@m3brown @marcesher @ssayre

### CODE REVIEW

**Comment by m3brown** [comment]
As long as you know that /var/log/newrelic/ is created before this particular job runs, this seems like a suitable solution.

**Comment by marcesher** [comment]
@sleitner @m3brown You could also just add a block to ensure that directory exists with the right permissions. That's safe and easy.

**Comment by sleitner** [comment]
@m3brown `file` acts like mkdir -p (as well as chown and chmod), so no worries about creating /var/log/newrelic/ first. @marcesher great! I was being overly cautious about modifying permissions, but I'm creating that directory with newrelic installs anyway.

**Comment by m3brown** [comment]
Agree with marc's idea about adjusting the permissions to /var/log/newrelic/.

We're being very overbearing here...but after working through the mysql requirements recently, I'm more conscious of the baseline compliance standards.  If you want to abide by those, `/var/log/newrelic/` should have 700 permissions (is apache is the only user using it?) and any log files inside that directory should be 660.

**Comment by sleitner** [comment]
Ah, so I should be paranoid about permissions in /var/log/ :). No problem. Apache and Root are the only users writing to /var/log/newrelic/*. I can modify permissions on the log files (660) and the directory (700). Anything else?

**Comment by sleitner** [comment]
Please take a minute to review this again, when you have the bandwidth..

**Comment by marcesher** [comment]
@sleitner 

My experience running this on production is that the log permission changes are not necessary and may in fact be not OK. 

Until we learn otherwise, I think we need to keep it the way it was

**Comment by sleitner** [comment]
@marcesher would you mind testing your alternate scenario in vagrant? I found previously that if I don't change permission for the apache user, it can't write to the logs.

And how were you running this in production? As far as I know everything we've done in prod pulled from my fork, which included these changes, so you'd have to revert permissions manually to test the cfpb version.

**Comment by marcesher** [comment]
@sleitner I ran from the cfpb org on the prod servers, where it didn't change permissions, and logging happens as expected.

File: provisioners/roles/newrelic_wsgi/tasks/main.yml [hunk 1]
```diff
@@ -9,6 +9,15 @@
     mode=755
     state=directory
   sudo: yes
+
+- name: create directory for newrelic logging accessible to apache user
+  file:
+    dest={{newrelic_agentlog_path}}
+    mode=700
```
Comment (marcesher) [inline provisioners/roles/newrelic_wsgi/tasks/main.yml]: keep this at 755. the new relic server daemon needs to be able to get into this directory, and 700 locks it down too tight

Comment (sleitner) [inline provisioners/roles/newrelic_wsgi/tasks/main.yml]: good catch :+1:

**Comment by marcesher** [comment]
Thanks @sleitner ! These work for me.

File: provisioners/roles/newrelic_php/tasks/main.yml [hunk 2]
```diff
@@ -41,7 +42,11 @@
   sudo: yes
 
 - name: modify php newrelic.daemon.auditlog
-  lineinfile: dest={{newrelic_ini_file}} state=present regexp="newrelic.daemon.auditlog =" line="newrelic.daemon.auditlog = /var/log/newrelic/audit.log"
+  lineinfile: dest={{newrelic_ini_file}} state=present regexp="newrelic.daemon.auditlog =" line="newrelic.daemon.auditlog = {{newrelic_agentlog_daemon}}"
```
Comment (marcesher) [inline provisioners/roles/newrelic_php/tasks/main.yml:45]: This should point to the newrelic_agentlog_daemonaudit var; I'll push a change for that

File: provisioners/roles/newrelic_php/tasks/main.yml [hunk 3]
```diff
@@ -41,7 +42,11 @@
...
+  sudo: yes
+
+- name: modify php newrelic.daemon.logfile
+  lineinfile: dest={{newrelic_ini_file}} state=present regexp="newrelic.daemon.logfile =" line="newrelic.daemon.logfile = /var/log/newrelic/newrelic-daemon.log"
```
Comment (marcesher) [inline provisioners/roles/newrelic_php/tasks/main.yml:49]: I'll replace this with newrelic_agentlog_daemon

File: provisioners/roles/newrelic_wsgi/defaults/main.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 # defaults file for newrelic_wsgi
 newrelic_ini_path: /opt/newrelic/
-newrelic_agentlog_path: /tmp
+newrelic_agentlog_path: /var/log/newrelic/
```
Comment (marcesher) [inline provisioners/roles/newrelic_wsgi/defaults/main.yml:4]: I'll remove the trailing slash; it causes double-slashes in the config files

**Pull request merged**